src/ResourcePlanner/Controllers/ReplanResourceController.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections.Generic;
+using System.Net;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Middleware.Common.Models;
+using Middleware.Common.Repositories;
+
+namespace Middleware.ResourcePlanner.Controllers
+{
+    [ApiController]
+    [Route("api/v1/[controller]")]
+    public class ReplanResourceController : ControllerBase
+    {
+        private readonly IResourcePlanner _resourcePlanner;
+        private readonly IActionPlanRepository _actionPlanRepository;
+        private readonly IMapper _mapper;
+        private readonly ILogger _logger;
+
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="resourcePlanner"></param>
+        /// <param name="mapper"></param>
+        /// <param name="logger"></param>
+        public ReplanResourceController(IResourcePlanner resourcePlanner, IMapper mapper, ILogger<ResourceController> logger)
+        {
+            _resourcePlanner = resourcePlanner;
+            _mapper = mapper;
+            _logger = logger;
+        }
+        public record ResourceInput(TaskModel Task, RobotModel Robot, bool FullReplan);
+
+        [HttpPost(Name = "GetResourceRePlan")]
+        [ProducesResponseType(typeof(TaskModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ApiResponse), (int)HttpStatusCode.NotFound)]
+        [ProducesResponseType(typeof(ApiResponse), (int)HttpStatusCode.InternalServerError)]
+
+        public async Task<ActionResult<TaskModel>> GetResourceRePlan([FromBody] ResourceInput resource)
+        {
+            try
+            {
+                //Lua query to get plan by robot and lastest timestamped.
+                List<ActionPlanModel> actionPlans = await _actionPlanRepository.GetActionPlanModelsAsync(resource.Robot.Id);
+
+                //Get the newest task of robot.
+                Dictionary<ActionPlanModel, DateTime> tempDic = new Dictionary<ActionPlanModel, DateTime>();
+                Dictionary<ActionPlanModel, DateTime> OrderedTempDic = new Dictionary<ActionPlanModel, DateTime>();
+
+                // Complete tempDic
+                foreach (ActionPlanModel plan in actionPlans)
+                {
+                    DateTime d;
+                    DateTime.TryParseExact(plan.Status, "ggyyyy$dd-MMM (dddd)", System.Globalization.CultureInfo.InvariantCulture, System.Globalization.DateTimeStyles.None, out d);
+                    tempDic.Add(plan, d);
+                }
+
+                // Order a new dictionary
+                foreach (KeyValuePair<ActionPlanModel, DateTime> pair in tempDic.OrderByDescending(p => p.Value))
+                {
+                    OrderedTempDic.Add(pair.Key, pair.Value);
+                }
+
+                // Get last item which is the latest plan.
+                ActionPlanModel last = OrderedTempDic.Keys.Last();
```

Comment:
When we are using `OrderByDescending` it orders the values from the biggest to lowest. So it means that the plan with the latest date will be first, because the time value from now is greater than the value of time from 2 days ago. You should use `First()` method.